hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -137,9 +137,10 @@ protected V getSyncInternal(Object key, ExpiryPolicy expiryPolicy) {
 
         try {
             long reservationId = nearCache.tryReserveForUpdate(key);
-            value = super.getSyncInternal(key, expiryPolicy);
+            Data keyData = toData(key);
```

Comment:
The key would anyway be serialized in `super.getSyncInternal()` and we also need the serialized key to create the `NearCacheRecord`. So we add the serialization here to have it in a single place again. The serialized key is then passed as additional argument to `tryPublishReserved()` which ends up in the `NearCacheRecordStore`.